File: 2021-remove-all-occurrences-of-a-substring/2021-remove-all-occurrences-of-a-substring.cpp
class Solution {
public:
    string removeOccurrences(string s, string part) {
        
        
        while(true){
            if(s.find(part)==string::npos){
                break;
            } else{
                int idx = s.find(part);
                s.erase(idx, part.length());
            }
        }
        return s;
    }
};

File: 2021-remove-all-occurrences-of-a-substring/2021-remove-all-occurrences-of-a-substring.java
class Solution {
    public String removeOccurrences(String s, String part) {
        
        while(true){
            if(!s.contains(part)){
                break;
            }else {
                int idx = s.indexOf(part);
                int start = idx;
                int end = idx + part.length();
                s = s.substring(0, start) + s.substring(end);
            }
            
        }
        return s;
    }
}